langroid/language_models/openai_gpt.py

```diff
@@ -309,37 +309,46 @@ def _process_stream_event(
             event = event.model_dump()
         event_args = ""
         event_fn_name = ""
-        if chat:
-            delta = event["choices"][0]["delta"]
-            if "function_call" in delta and delta["function_call"] is not None:
-                if "name" in delta["function_call"]:
-                    event_fn_name = delta["function_call"]["name"]
-                if "arguments" in delta["function_call"]:
-                    event_args = delta["function_call"]["arguments"]
-            event_text = delta.get("content", "")
-        else:
-            event_text = event["choices"][0]["text"]
-        if event_text:
-            completion += event_text
-            if not is_async:
-                sys.stdout.write(Colors().GREEN + event_text)
-                sys.stdout.flush()
-        if event_fn_name:
-            function_name = event_fn_name
-            has_function = True
-            if not is_async:
-                sys.stdout.write(Colors().GREEN + "FUNC: " + event_fn_name + ": ")
-                sys.stdout.flush()
-        if event_args:
-            function_args += event_args
-            if not is_async:
-                sys.stdout.write(Colors().GREEN + event_args)
-                sys.stdout.flush()
-        if event["choices"][0].get("finish_reason", "") in ["stop", "function_call"]:
-            # for function_call, finish_reason does not necessarily
-            # contain "function_call" as mentioned in the docs.
-            # So we check for "stop" or "function_call" here.
-            return True, has_function, function_name, function_args, completion
+        event_text = ""
```

Comment:
this makes it look like there are more changes than there actually are.

instead of surrounding this while code with an `if` clause, there's a better way to do this that will make the changes clearer. Left as an exercise :)

Reply:
How about this?